job_crafter.py

```diff
@@ -102,75 +111,96 @@ def make_jobs(self):
 
         self.job["device_type"] = self.board['device_type']
 
-        if self.options['default_notify']:
+        if self.cfg['default_notify']:
             self.job["notify"] = self.board.get("notify", [])
         else:
-            self.job["notify"] = self.options["notify"]
-        print("notify recipients: Overridden")
+            self.job["notify"] = self.cfg['notify']
+
+        logging.info("Notifications recipients: %s" % ", ".join(self.job['notify']))
 
         # Define which test to run
         tests = []
-        if self.options['tests']:
-            tests = [next(iter([e for e in self.board['tests'] if e['name'] == t]), {'name': t}) for t in self.options['tests']]
+        if 'tests' in self.cfg:
+            # WTF?
```

Comment:
This is needed to generate the test list as a list of dictionaries, as in the `boards.py` file.  

It has two purposes:
  * When a test is in not affected to a board in `boards.py`, this simulates it in order to have the same data structure for the rest of the code. (a list of dict, and not a list of strings)
  * When a test has been affected to a board, but overrides a default value (the template to use, for example; useful for multinode tests), this makes sure to take that value into account by transforming the test name given in the command line, by the dictionary of the configuration.

This surely would need a comment.

BTW, I think these lines are now the only path, since `--tests` is now set as mandatory there: https://github.com/free-electrons/custom_tests_tool/pull/1/files?diff=unified#r127648155